carapace-server/src/main/java/org/carapaceproxy/server/mapper/StandardEndpointMapper.java

```diff
@@ -276,13 +277,22 @@ public SimpleHTTPResponse mapInternalError(String routeid) {
 
     @Override
     public SimpleHTTPResponse mapServiceUnavailableError(String routeId) {
-        // custom global
-        if (defaultServiceUnavailable != null) {
-            ActionConfiguration errorAction = actions.get(defaultServiceUnavailable);
-            if (errorAction != null) {
-                return new SimpleHTTPResponse(errorAction.getErrorCode(), errorAction.getFile(), errorAction.getCustomHeaders());
+        ActionConfiguration errorAction = null;
+        // custom for route
+        Optional<RouteConfiguration> config = routes.stream().filter(r -> r.getId().equalsIgnoreCase(routeId)).findFirst();
+        if (config.isPresent()) {
+            String action = config.get().getErrorAction();
+            if (action != null) {
+                errorAction = actions.get(action);
             }
         }
+        // custom global
+        if (errorAction == null && defaultServiceUnavailable != null) {
+            errorAction = actions.get(defaultServiceUnavailable);
+        }
+        if (errorAction != null) {
+            return new SimpleHTTPResponse(errorAction.getErrorCode(), errorAction.getFile(), errorAction.getCustomHeaders());
+        }
```

Comment:
This part is pretty much identical in:

- `mapInternalError`: lines 258-273
- `mapServiceUnavailableError`: lines 280-295

Maybe we should group it in an external method?